### DISCUSSION TITLE
Allow dynamic meshes to sample LightmapGI probes from custom location

### DISCUSSION THREAD

**Discussion by jadeitejt** [godotengine/godot-proposals#10450] [3D]
Sometimes it's useful to manually set the location a mesh will use to sample baked lighting. I ran into this issue making a set of double doors, where ideally both doors would sample from the center of the doorway and have the same lighting. I already tried to do this by changing the origin and it didn't seem to do anything.

![capture](https://github.com/user-attachments/assets/713ec5fc-26b4-4b54-91de-4cd69dc9ca35)

The page for a similar source engine feature also describes the use case of large meshes embedded in world geometry 
[https://developer.valvesoftware.com/wiki/Info_lighting](https://developer.valvesoftware.com/wiki/Info_lighting)

Something like this maybe?

![mockup](https://github.com/user-attachments/assets/9d8cfe42-7ac6-43da-a4fd-35096fd045d6)

**Comment by Calinou** [comment]
This makes sense to me, but are you referring to LightmapGI probes by "gi probes"? When using VoxelGI or SDFGI, dynamic objects sample GI data with the same resolution as static objects (so they blend in perfectly if they wouldn't contribute GI otherwise).

As a workaround, modify your mesh data so it's offset from the origin. For OBJ as Mesh imports, there is an Offset Mesh import option you can use (no such option exists for 3D scene import though).

**Reply by jadeitejt** [reply]
Sorry, yes, I do mean the LightmapGI probes. I've tried offsetting the mesh from the origin and it doesn't work, it seems like sampling is done center mass.

Here's a mesh offset from its origin rotating in place from a dark area to a light area. Dynamic lights are off, so it's just baked lighting influence

![offset](https://github.com/user-attachments/assets/9d7771c8-1330-407c-9632-3afd74455c46)
